cmd/prometheus/main.go

```diff
@@ -323,6 +323,9 @@ func main() {
 	serverOnlyFlag(a, "storage.tsdb.retention.size", "Maximum number of bytes that can be stored for blocks. A unit is required, supported units: B, KB, MB, GB, TB, PB, EB. Ex: \"512MB\". Based on powers-of-2, so 1KB is 1024B.").
 		BytesVar(&cfg.tsdb.MaxBytes)
 
+	serverOnlyFlag(a, "storage.tsdb.retention.percentage", "Maximum percentage of the disk space that can be used to store blocks (prevails over storage.tsdb.retention.size).").
```

Comment:
On first reading I interpreted "disk space" to mean "free space"; after looking more deeply I realize you implemented as a percentage of the total disk (partition) size.
Also "can be used to store blocks" might suggest that the head (WAL, chunks, etc.) is not included, whereas I believe it is.

Suggest to make this clearer; maybe add some commentary in the docs.

Reply:
I can change the message, no problem. What would you suggest ?